Storage/SQL/SQLiteLogins.swift

```diff
@@ -566,12 +561,19 @@ public class SQLiteLogins: BrowserLogins {
         "SELECT guid, \(nowMillis), 1, \(SyncStatus.Changed.rawValue), '', timeCreated, \(nowMillis)000, '', '' FROM \(TableLoginsMirror) WHERE guid IN \(inClause)"
 
         let args: Args = guids.map { $0 as AnyObject }
-        return self.db.run([
+        return [
             (delete, args),
             (update, args),
             (markMirrorAsOverridden, args),
             (insert, args)
-        ]) >>> effect(self.notifyLoginDidChange)
+        ]
+    }
+
+    public func removeLoginsWithGUIDs(guids: [GUID]) -> Success {
+        let timestamp = NSDate.now()
+        return db.run(chunk(guids, by: BrowserDB.MaxVariableNumber).flatMap {
+            self.getDeletionStatementsForGUIDs(Array($0), nowMillis: timestamp)
```

Comment:
You should be able to omit the `Array` call if you just make `getDeletionStatementsForGUIDs` take `ArraySlice<GUID>` instead of `[GUID]`. No point doing all of those extra allocations when you control the private method.